File: src/main/java/com/zestarr/pluginportal/utils/ConfigUtils.java
package com.zestarr.pluginportal.utils;

import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

public class ConfigUtils {

    public static File getPluginFolder() {
        return new File("plugins");
    }

    public static File getDebugPluginFolder() {
        return new File("plugins" + File.separator + "DebugPlugins");
    }

    public static File getPluginListFile() {
        return new File("PluginPortalPlugins.yml");
    }

    public static YamlConfiguration getPluginListFileConfig() {
        return YamlConfiguration.loadConfiguration(getPluginListFile());
    }

    public static File createPluginDataFile() {
        File file = new File("PluginData.json");
        try {
            if (!file.exists()) {
                file.createNewFile();
            }
        } catch (IOException exception) {
            exception.printStackTrace();
        }

        return file;
    }

}

File: src/main/java/com/zestarr/pluginportal/utils/ConfigUtils.kt
package com.zestarr.pluginportal.utils

import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

object ConfigUtils {
    val pluginFolder: File
        get() = File("plugins")
    val pluginListFile: File
        get() = File("Plugins.yml")
    val pluginListFileConfig: YamlConfiguration
        get() = YamlConfiguration.loadConfiguration(pluginListFile)
}